### PR TITLE
Added swipe to share feature

### PR DESCRIPTION
**Pull request by ivanlares**

### CODE REVIEW

File: VoiceMemos/Controller/VoicesTableViewController.swift [hunk 1]
```diff
@@ -310,10 +293,47 @@ class VoicesTableViewController: BaseTableViewController, UISearchBarDelegate {
         }
     }
     
-    override func tableView(tableView: UITableView, editingStyleForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCellEditingStyle {
-        return .Delete
+    override func tableView(tableView: UITableView, editActionsForRowAtIndexPath indexPath: NSIndexPath) -> [UITableViewRowAction]? {
+        
+        let deleteAction = UITableViewRowAction(style: .Normal, title: "Delete"){ action, index in
+            
+            dispatch_async(dispatch_get_main_queue()){
```
Comment (MoZhouqi) [inline VoiceMemos/Controller/VoicesTableViewController.swift]: The delegate method is executed on the main thread, so the `dispatch_async` call is unnecessary.

File: VoiceMemos/Controller/VoicesTableViewController.swift [hunk 2]
```diff
@@ -310,10 +293,47 @@ class VoicesTableViewController: BaseTableViewController, UISearchBarDelegate {
         }
     }
     
-    override func tableView(tableView: UITableView, editingStyleForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCellEditingStyle {
-        return .Delete
+    override func tableView(tableView: UITableView, editActionsForRowAtIndexPath indexPath: NSIndexPath) -> [UITableViewRowAction]? {
+        
+        let deleteAction = UITableViewRowAction(style: .Normal, title: "Delete"){ action, index in
```
Comment (MoZhouqi) [inline VoiceMemos/Controller/VoicesTableViewController.swift]: Need a space before the handler block.

File: VoiceMemos/Controller/VoicesTableViewController.swift [hunk 3]
```diff
@@ -310,10 +293,47 @@ class VoicesTableViewController: BaseTableViewController, UISearchBarDelegate {
...
+        
+        let shareAction = UITableViewRowAction(style: .Normal, title: "Share") { action, index in
+            
+            dispatch_async(dispatch_get_main_queue()){
+                
+                let voice = self.voiceForRowAtIndexPath(indexPath, WithTableView: tableView)
+                let audioPath = self.directoryURL.URLByAppendingPathComponent(voice.filename!)
+                let activityView = UIActivityViewController(activityItems: [audioPath], applicationActivities: nil)
+                let currentCell = tableView.cellForRowAtIndexPath(indexPath)
+                activityView.popoverPresentationController?.sourceView = tableView.cellForRowAtIndexPath(indexPath)
```
Comment (MoZhouqi) [inline VoiceMemos/Controller/VoicesTableViewController.swift]: Could use the `currentCell`.

File: VoiceMemos/Controller/VoicesTableViewController.swift [hunk 4]
```diff
@@ -310,10 +293,47 @@ class VoicesTableViewController: BaseTableViewController, UISearchBarDelegate {
...
+                self.presentViewController(activityView, animated: true, completion: nil)
+                
+            }
+            
+        }
+        
+        deleteAction.backgroundColor = UIColor.redColor()
+        shareAction.backgroundColor = UIColor.blueColor()
+        
+        return [shareAction, deleteAction]
```
Comment (MoZhouqi) [inline VoiceMemos/Controller/VoicesTableViewController.swift]: May be `[deleteAction, shareAction]` is better and you could also change the order of the `UITableViewRowAction` initializers.

File: VoiceMemos/Controller/VoicesTableViewController.swift [hunk 5]
```diff
@@ -310,10 +293,47 @@ class VoicesTableViewController: BaseTableViewController, UISearchBarDelegate {
...
+                }
+        
+                self.deleteVoiceInPersistentStore(voice)
+            }
+            
+        }
+        
+        let shareAction = UITableViewRowAction(style: .Normal, title: "Share") { action, index in
+            
+            dispatch_async(dispatch_get_main_queue()){
```
Comment (MoZhouqi) [inline VoiceMemos/Controller/VoicesTableViewController.swift]: As Above, the `dispatch_async` call is unnecessary.

File: VoiceMemos/Controller/VoicesTableViewController.swift [hunk 6]
```diff
@@ -310,10 +293,47 @@ class VoicesTableViewController: BaseTableViewController, UISearchBarDelegate {
...
     }
     
-    override func tableView(tableView: UITableView, editingStyleForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCellEditingStyle {
-        return .Delete
+    override func tableView(tableView: UITableView, editActionsForRowAtIndexPath indexPath: NSIndexPath) -> [UITableViewRowAction]? {
+        
+        let deleteAction = UITableViewRowAction(style: .Normal, title: "Delete"){ action, index in
+            
+            dispatch_async(dispatch_get_main_queue()){
+                let voice = self.voiceForRowAtIndexPath(indexPath, WithTableView: tableView)
```
Comment (MoZhouqi) [inline VoiceMemos/Controller/VoicesTableViewController.swift]: The handler block of the share action is also use the `voice` property, so you could consider to put this line of code at the beginning of the delegate method.

**Comment by MoZhouqi** [comment]
Thanks for your awesome pull request, but there are a few problems, could you please fix it?

File: VoiceMemos/Controller/VoicesTableViewController.swift (hunk 4 above)
Comment (ivanlares) [inline VoiceMemos/Controller/VoicesTableViewController.swift]: I agree `[deleteAction, shareAction]` is better but the order or the initializers is already chronological. 

```
deleteAction = ...
shareAction = ... 
```

Should I initialize shareAction first ?

**Comment by ivanlares** [comment]
Thanks for the feedback. 
I'm going to add a second commit to the topic_branch, should I squash the commits? 
Can you squash the commits, I'm not sure what is best.

File: VoiceMemos/Controller/VoicesTableViewController.swift [hunk 7]
```diff
@@ -310,10 +293,35 @@ class VoicesTableViewController: BaseTableViewController, UISearchBarDelegate {
         }
     }
     
-    override func tableView(tableView: UITableView, editingStyleForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCellEditingStyle {
-        return .Delete
+    override func tableView(tableView: UITableView, editActionsForRowAtIndexPath indexPath: NSIndexPath) -> [UITableViewRowAction]? {
+        let voice = self.voiceForRowAtIndexPath(indexPath, WithTableView: tableView)
+        
+        let deleteAction = UITableViewRowAction(style: .Normal, title: "Delete"){ action, index in
```
Comment (MoZhouqi) [inline VoiceMemos/Controller/VoicesTableViewController.swift]: Need a space before the handler block.

**Comment by MoZhouqi** [comment]
Thanks! You can meld all commits into one, then I will merge the pull request.

File: VoiceMemos/Controller/VoicesTableViewController.swift (hunk 7 above)
Comment (ivanlares) [inline VoiceMemos/Controller/VoicesTableViewController.swift]: I'm confused by the comment. 
Should it be like this ? 

```
let deleteAction = UITableViewRowAction(style: .Normal, title: "Delete"){ action, index in
SPACE_HERE
CODE_HERE
```

**Edit:**
I belive this is what you meant

```
let deleteAction = UITableViewRowAction(style: .Normal, title: "Delete") { action, index in
```

Comment (MoZhouqi) [inline VoiceMemos/Controller/VoicesTableViewController.swift]: Yes, you are right.

**Pull request merged**